src/dbnode/storage/namespace.go

```diff
@@ -393,6 +397,18 @@ func (n *dbNamespace) Shards() []Shard {
 	return databaseShards
 }
 
+func (n *dbNamespace) SchemaRegistry() namespace.SchemaRegistry {
+	n.RLock()
+	defer n.RUnlock()
```

Comment:
nit: Avoid defer here since it's easy to do and may be called in the fast path for a write?